fury/actor.py

```diff
@@ -1545,16 +1545,27 @@ def sphere(centers, colors, radii=1., phi=16, theta=16,
     >>> # window.show(scene)
 
     """
-    src = SphereSource() if faces is None else None
+    if not prim:
+        src = SphereSource() if faces is None else None
 
-    if src is not None:
-        src.SetRadius(1)
-        src.SetThetaResolution(theta)
-        src.SetPhiResolution(phi)
+        if src is not None:
+            src.SetRadius(1)
+            src.SetThetaResolution(theta)
+            src.SetPhiResolution(phi)
 
-    actor = repeat_sources(centers=centers, colors=colors,
-                           active_scalars=radii, source=src,
-                           vertices=vertices, faces=faces)
+        actor = repeat_sources(centers=centers, colors=colors,
+                               active_scalars=radii, source=src,
+                               vertices=vertices, faces=faces)
+
+    else:
+        scales = np.multiply(radii, [1, 1, 1])
+        directions = (1, 0, 0)
+        verts, faces = fp.prim_sphere('symmetric724', False)
+        res = fp.repeat_primitive(verts, faces,
+                                  directions=directions, centers=centers,
+                                  colors=colors, scales=scales)
+        big_verts, big_faces, big_colors, _ = res
+        actor = get_actor_from_primitive(big_verts, big_faces, big_colors)
```

Comment:
In this case, `phi` and `theta` and `opacity` are not used which is not good. 

- You will have to update the `prim_sphere` function to generate the point of a sphere following `phi` and `theta`
- You will have to manage the opacity also.

Reply:
Thanks @skoudoro, I will be changing the `prim_sphere` function to use `phi` and `theta`. However, the opacity of the sphere is managed by the following line :
```py
actor.GetProperty().SetOpacity(opacity)
```
And opacity can also be set through colors parameter by passing (R,G,B,A) values

Reply:
Indeed,  good catch, so forgets the opacity and focus just on the updating `prim_shere`.

Reply:
In the `prim_sphere` method, it gets `vertices` and `faces` from`.npz` file and akes `name` and `gen_faces` parameters, should I replace it or make it optional to use `name` and `gen_faces` parameter.

Reply:
you should keep it, it is really important

Reply:
Okay

Reply:
> I noticed faces of the sphere are quadrilateral, Is it fine to go with?

and yes, it is fine. As an option, we should be able to increase the number of points (density) and the quadrilateral will be smaller.

Reply:
Hi @skoudoro, made the required changes to the `prim_sphere`. Can you please have a look?